api/src/main/java/com/valtech/aem/saas/api/typeahead/TypeaheadService.java

```diff
@@ -12,10 +14,13 @@
   /**
    * Retrieves typeahead results
    *
-   * @param index               SaaS client index.
-   * @param typeaheadPayloadDto object containing typeahead query values.
+   * @param searchConfiguration sling model accessing context aware search configurations (i.e client and index).
+   * @param text                search term.
+   * @param language            search language scope.
+   * @param filters             search filters
    * @return List of string represented typeahead options. Empty list if no options are found.
    */
-  List<String> getResults(@NonNull String index, @NonNull TypeaheadPayloadDTO typeaheadPayloadDto);
+  List<String> getResults(@NonNull SearchCAConfigurationModel searchConfiguration, @NonNull String text,
+      @NonNull String language, Set<FilterModel> filters);
```

Comment:
double-checking: the filters are additional filters to the ones in the searchConfiguration.  correct?

Reply:
exactly